### DISCUSSION TITLE
Constant "reinstalling dependents with broken linkage from source" for php and httpd

### DISCUSSION THREAD

**Discussion by karlshea** [Homebrew/discussions#3218] [Everyday usage] [answered]
On two different Macs Homebrew is *constantly* trying to recompile php and httpd when I do a `brew upgrade`. Example from tonight:

```
==> Upgrading 3 outdated packages:
mkcert 1.4.3 -> 1.4.4
ruby-build 20220415 -> 20220426
ca-certificates 2022-03-29 -> 2022-04-26
...
==> No outdated dependents to upgrade!
==> Checking for dependents of upgraded formulae...
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Reinstalling 2 dependents with broken linkage from source:
httpd, php
```

If I `brew remove httpd`, `brew remove php`, `brew cleanup`, then `brew install httpd`, `brew install php` it pours the bottles normally.

These are the only two packages that are behaving this way, is there something I can do to troubleshoot this?

**brew config**
```
HOMEBREW_VERSION: 3.4.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: d0a0bbef8db45cca303c607037338d02d8eaa8ca
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c8a327ce86e7dd61c434855528c6d9a96d0237f8
Core tap last commit: 49 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 13.1.6 build 1316
Git: 2.36.0 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-x86_64
CLT: 13.3.1.0.1.1648687083
Xcode: 13.3.1
```

**brew doctor**
```
Your system is ready to brew.
```

**Comment by cho-m** [comment, marked as answer]
If I were to guess, it may be the known `apr-util` issue:
- Homebrew/homebrew-core#96892
- Homebrew/homebrew-core#99400
- Homebrew/homebrew-core#100189


You can troubleshoot by running `brew linkage httpd` or `brew linkage php` when you see this type of issue to find which formula's link is broken.

Once you identify problematic formulae, you might be able to fix issue by `brew reinstall ...`

In case of `apr-util` issue, we plan to fix it after merging `libxcrypt` (Homebrew/homebrew-core#99367) and then rebuilding+distributing revision-bumped `apr-util`

**Reply by karlshea** [reply]
Looks like that apr fix went in a bit after you posted, and since then I've had no issues with this.
